src/openrct2/peep/peep.c

```diff
@@ -9795,6 +9795,9 @@ int peep_pathfind_choose_direction(sint16 x, sint16 y, uint8 z, rct_peep *peep)
 				/* Peep remembers this junction, so remove the
 				 * chosen_edge from those left to try. */
 				peep->pathfind_history[i].direction &= ~(1 << chosen_edge);
+				/* Also remove the edge the peep came from from
+				 * those left to try. */
```

Comment:
@Broxzier Yes, that's fair enough. I'm happy to change it to reduce potential confusion.
@janisozaur Thanks for the suggestions.